course_catalog/views.py

```diff
@@ -328,3 +339,40 @@ class TopicViewSet(viewsets.ReadOnlyModelViewSet):
     serializer_class = CourseTopicSerializer
     pagination_class = LargePagination
     permission_classes = (AnonymousAccessReadonlyPermission,)
+
+
+class WebhookOCWView(APIView):
+    """
+    Handle webhooks coming from the OCW bucket
+    """
+
+    permission_classes = ()
+    authentication_classes = ()
+
+    def handle_exception(self, exc):
+        """Raise any exception with request info instead of returning response with error status/message"""
+        raise WebhookException(
+            "REST Error (%s). BODY: %s, META: %s"
+            % (exc, self.request.body, self.request.META)
+        ) from exc
+
+    def post(self, request):
+        """Process webhook request"""
+        content = rapidjson.loads(request.body.decode())
+        records = content.get("Records")
+        if features.is_enabled(features.WEBHOOK_OCW) and records is not None:
+            for record in content.get("Records"):
+                s3_key = record.get("s3", {}).get("object", {}).get("key")
```

Comment:
This should end with `.get("key", "")` so that the `split` on the next line will work

Reply:
I did it this to way purposely so an exception would be raised on the next line if it's not there - if there is no S3 key, it's not a valid request (tested via sending `{"Records": [{"foo": "bar"}`)